Add `atty` potential unsoundness (unaligned read) advisory (#1462)
rustsec/advisory-db: crates/atty/RUSTSEC-0000-0000.md

```toml
[advisory]
id = "RUSTSEC-0000-0000"
package = "atty"
date = "2021-07-04"
url = "https://github.com/softprops/atty/issues/50"
categories = ["unaligned-read"]
informational = "unsound"

[affected]
os = ["windows"]
```

> 

Reviewer: ```suggestion
[versions]
patched = []

```

Needs this - it would be nice if the maintainer merges the fix before we merge this

Author: Yeah, would be nice, but given that the last commit to `atty` was made more than a year ago and has many open PRs fixing issues (including this soundness issue), it seems unlikely to happen soon.

# Potential unaligned read

On windows, `atty` dereferences a potentially unaligned pointer.

> In practise however, the pointer won't be unaligned unaligned unless a custom global allocator is used.

Reviewer: ```suggestion
In practice however, the pointer won't be unaligned unless a custom global allocator is used.
```

> In particular, the `System` allocator on windows uses `HeapAlloc`, which guaranties large enough alignment.

Reviewer: ```suggestion
In particular, the `System` allocator on windows uses `HeapAlloc`, which guarantees a large enough alignment.
```

> ## References

Reviewer: We can remove these references here as it gets rendered via `url = []` and `references = []` - thanks :)
